sources/updater/update_metadata/src/lib.rs

```diff
@@ -320,34 +319,40 @@ impl Update {
         }
     }
 
-    pub fn update_ready(&self, seed: u32) -> bool {
-        // Has this client's wave started
-        if let Some(wave) = self.update_wave(seed) {
-            return wave.has_started();
-        }
-
-        // Or there are no waves
-        true
-    }
-
-    pub fn jitter(&self, seed: u32) -> Option<DateTime<Utc>> {
+    /// Returns whether the update is truly available. A update is said to be 'ready/available' if the wave
+    /// this host belongs to has fully passed, or if the host's position in the wave has passed.
+    /// The position of the host within the wave is determined by the seed value.
+    pub fn update_ready(&self, seed: u32, time: DateTime<Utc>) -> bool {
+        // If this host is part of some update wave
         if let Some(wave) = self.update_wave(seed) {
             if wave.has_passed() {
-                return None;
+                return true;
             }
-            let bounds = match self.update_wave(seed) {
-                Some(Wave::Initial { end }) => Some((Utc::now(), end)),
-                Some(Wave::General { start, end }) => Some((start, end)),
-                Some(Wave::Last { start: _ }) | None => None,
+            let bound = match wave {
+                // For all intent and purposes, with the way we define update wave structures,
+                // the initial wave (0th wave) never has any updates.
+                // No seed value would ever land the host in this initial wave. But if for whatever reason
+                // this host is part of the initial (0th) wave, the update is going to be available
```

Comment:
```suggestion
                // Hosts should not wind up in the special "initial" wave with no start time, but if they do,
                // we consider the update as being available immediately.
```
I'm not sure I understand the initial wave timing, and it feels like something we're liable to change (I'd really like to get rid of the unused special case) so I'd prefer to simplify this explanation.